addons/website_portal_purchase/controllers/main.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import OrderedDict
+
+from odoo import http
+from odoo.exceptions import AccessError
+from odoo.http import request
+from odoo.tools.translate import _
+
+from odoo.addons.website_portal.controllers.main import website_account, get_records_pager
+
+
+class WebsiteAccount(website_account):
```

Comment:
I know we are crazy. But could'nt we put this definition in a website_portal.py file, and name the class WebsitePortal ?

Yo crazy moma !